goshopify.go

```diff
@@ -211,10 +211,18 @@ func CheckResponseError(r *http.Response) error {
 		Errors interface{} `json:"errors"`
 	}{}
 
-	decoder := json.NewDecoder(r.Body)
-	err := decoder.Decode(&shopifyError)
-	if err != nil {
-		return err
+	if r.StatusCode == 406 {
```

Comment:
I think it would be better to simply check if the body of the request (`r.Body`) is empty and then stop decoding if that's the case, adding a generic error message like "unknown error", or create another special case [here](https://github.com/getconversio/go-shopify/blob/366c7fd2721c43eb84c184bfdf85a2a54fe19dfc/goshopify.go#L192-L201).

Reply:
you bet I'll make those changes and add some tests